static/app/components/autoComplete.tsx

```diff
@@ -400,12 +411,12 @@ class AutoComplete<T extends Item> extends Component<Props<T>, State<T>> {
 
     onClose?.(...args);
 
-    if (this.isControlled || !this._mounted) {
+    if (!this._mounted) {
```

Comment:
Removing the `isControlled` early return here is what fixes `resetInputOnClose` not working when `isOpen` is controlled but `inputValue` is not. There's a test in `autoComplete.spec.jsx` that validates this behavior